gradle-plugin-release-plugin/plugin/src/main/groovy/com/novoda/releaseplugin/internal/ParseChangeLogTask.groovy

```diff
@@ -0,0 +1,47 @@
+package com.novoda.releaseplugin.internal
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+class ParseChangeLogTask extends DefaultTask {
+
+    @Input
+    final Property<String> changelog
+
+    @Input
+    final Property<String> version
+
+    @OutputFile
+    File releaseChangelog
+
+    ParseChangeLogTask() {
+        group = 'changelog'
+        description = 'Extracts the changelog for the current release'
+        changelog = project.objects.property(String)
+        version = project.objects.property(String)
+    }
+
+    @TaskAction
+    void run() {
+        def changelog = extractChangelog()
+        releaseChangelog.text = changelog
+    }
+
+    String extractChangelog() {
+        String fullChangelog = new File(changelog.get()).text
+        def latestChangelog = (fullChangelog =~ /\[Version ${version.get()}.*\n-*([\s\S]*?)\[Version.*\n-*/)
+        if (latestChangelog.size() > 0) {
```

Comment:
Hmm, the two `if` statements look pretty similar. I wonder if there would be a matcher method to make this more clear. What's that regex about anyway? (might be worth a comment)